plans/upgrade_mongodb.pp

```diff
@@ -0,0 +1,100 @@
+# - set MongoDB feature compatibility to 3.4
+# - change Yum repo to 3.6
+# - upgrade packages
+# - set MongoDB feature compatibility to 3.6
+# ## https://docs.mongodb.com/manual/release-notes/4.0-upgrade-standalone/
+# - change Yum repo to 4.0
+# - upgrade packages
+# - set MongoDB feature compatibility to 4.0
+# - start stackstorm
+#
+# @param [TargetSpec] targets
+#   Set of targets (MongoDB hosts) that this plan will be executed on.
+#
+# @param [String] mongo_admin_db
+#   Name of the admin database for MongoDB
+#
+# @param [String] mongo_username
+#   Name of the admin user on the admin database
+#
+# @param [String] mongo_password
+#   Password of the admin user on the admin database
+#
+# @param [Array[String]] mongo_packages
+#   List of MongoDB packages that will be upgraded
+#
+# @param [String] upgrade_version_start
+#   Version of MongoDB that the database is currently on, ie. where we are starting from.
+#
+# @param [Array[String]] upgrade_version_path
+#   List of versions that we will upgrade through along our path to success!
+#
+# @example Basic usage
+#   bolt plan run st2::upgrade_mongodb --targets ssh_nodes --params '{"mongo_password": "xxx"}'
+#
+# @example Upgrading enterprise packages
+#   bolt plan run st2::upgrade_mongodb --targets ssh_nodes --params '{"mongo_password": "xxx", "mongo_packages": ["mongodb-enterprise-server", "mongodb-enterprise-shell", "mongodb-enterprise-tools"]}'
+#
+# @example Upgrading from 3.6 to 4.0
+#   bolt plan run st2::upgrade_mongodb --targets ssh_nodes --params '{"mongo_password": "xxx", "upgrade_version_start": "3.6", "upgrade_version_path": ["4.0"]}'
+#
+# @example Upgrading from 3.4 to 3.6 to 4.0
+#   bolt plan run st2::upgrade_mongodb --targets ssh_nodes --params '{"mongo_password": "xxx", "upgrade_version_start": "3.4", "upgrade_version_path": ["3.6", "4.0"]}'
+#
+plan st2::upgrade_mongodb (
+  String $mongo_admin_db = 'admin',
+  String $mongo_username = 'admin',
+  String $mongo_password,
+  Array[String] $mongo_packages  = ['mongodb-org-server', 'mongodb-org-shell', 'mongodb-org-tools'],
+  String $upgrade_version_start = '3.4',
+  Array[String] $upgrade_version_path = ['3.6', '4.0'],
+  TargetSpec $targets,
+) {
+  # stop stackstorm
+  run_command('st2ctl stop', $targets)
+
+  $mongo_cmd = "mongo ${mongo_admin_db} --username ${mongo_username} --password ${mongo_password} --quiet"
+
+  # set MongoDB feature compatibility to 3.4
+  $start_ver = $upgrade_version_start
+  run_command("$mongo_cmd --eval \"db.adminCommand( { setFeatureCompatibilityVersion: '$start_ver' } )\"",
+              $targets,
+              "Mongodb - Set Feature Compatibility Version $start_ver")
+
+  $upgrade_version_path.each |$ver| {
+    # Chnage Yum repo to this version
+    apply($targets) {
+      yumrepo { 'mongodb':
+        descr    => 'MongoDB Repository',
+        baseurl  => "https://repo.mongodb.org/yum/redhat/\$releasever/mongodb-org/${ver}/\$basearch/",
```

Comment:
If this is enterprise mongo then wouldn't it be mongodb-enterprise instead of mongodb-org here?

If we support the packages listing the enterprise names, then should we also let the org be configurable so its mongodb-org or mongodb-enterprise (or derive it from the package names)